pilot/pkg/model/config.go

```diff
@@ -58,12 +57,15 @@ type ConfigKey struct {
 }
 
 func (key ConfigKey) HashCode() ConfigHash {
-	hash := md5.New()
-	hash.Write([]byte{byte(key.Kind)})
-	hash.Write([]byte(key.Name))
-	hash.Write([]byte(key.Namespace))
-	sum := hash.Sum(nil)
-	return ConfigHash(binary.BigEndian.Uint64(sum))
+	var builder strings.Builder
```

Comment:
why use a String if we convert to a Byte latter? isn't it cheaper to use a `[]byte` directly?

Reply:
Why not use the same approach we're already using? You can use `xxhash.New()` and the rest should be the same.

Reply:
You can also do `digest.WriteString()` you avoid converting to byte arrays when writing to the digest

Reply:
Thanks for reminding me. That's fine, and I've committed the changes.

new benchmark
```code
name                      old time/op    new time/op    delta
HashCode/small_string-4      197ns ±13%      46ns ± 8%   -76.79%  (p=0.000 n=9+10)
HashCode/middle_string-4     298ns ±14%      66ns ±15%   -77.72%  (p=0.000 n=10+10)
HashCode/long_string-4       538ns ±20%     229ns ±39%   -57.49%  (p=0.000 n=10+10)

name                      old alloc/op   new alloc/op   delta
HashCode/small_string-4      16.0B ± 0%      0.0B       -100.00%  (p=0.000 n=10+10)
HashCode/middle_string-4     16.0B ± 0%      0.0B       -100.00%  (p=0.000 n=10+10)
HashCode/long_string-4        160B ± 0%      144B ± 0%   -10.00%  (p=0.000 n=10+10)

name                      old allocs/op  new allocs/op  delta
HashCode/small_string-4       1.00 ± 0%      0.00       -100.00%  (p=0.000 n=10+10)
HashCode/middle_string-4      1.00 ± 0%      0.00       -100.00%  (p=0.000 n=10+10)
HashCode/long_string-4        3.00 ± 0%      2.00 ± 0%   -33.33%  (p=0.000 n=10+10)
```